### PR TITLE
feat: mr support custom upstream/target branch

### PR DESCRIPTION
**Pull request by onlysuncolour**

### CODE REVIEW

**Comment by whatwewant** [comment]
#18

**Review by ppxu — APPROVED** [review]

File: core/mr [hunk 1]
```diff
@@ -3,40 +3,27 @@
...
-  local git_repo_url=$(git config --get remote.origin.url)
+  local git_repo_url=$(git::repo_url)
   # http://10.27.249.150:8888/thinking-frontend/ta-multiverse
   # http://10.27.249.150:8888/thinking-frontend/ta-multiverse.git
-  local git_repo_full_name=$(echo $git_repo_url |awk -F '10.27.249.150:8888/' '{print $2}')
-  git_repo_full_name=$(echo $git_repo_full_name |awk -F '.git' '{print $1}')
-  if [ "$git_repo_full_name" = "" ]; then
+  # ssh://git@139.224.169.199:26662/thinking-frontend/ta-multiverse.git
+
+  local checkHttpUrl=$(echo $git_repo_url |awk -F '10.27.249.150:8888/' '{print $2}')
```
Comment (whatwewant) [inline core/mr:12]:
```suggestion
  local check_http_url=$(echo $git_repo_url | awk -F '10.27.249.150:8888/' '{print $2}')
```

File: core/mr [hunk 2]
```diff
@@ -3,40 +3,27 @@
...
+  local checkSshUrl=$(echo $git_repo_url |awk -F '139.224.169.199:26662/' '{print $2}')
```
Comment (whatwewant) [inline core/mr:13]:
```suggestion
  local check_ssh_url=$(echo $git_repo_url |awk -F '139.224.169.199:26662/' '{print $2}')
```

File: core/mr [hunk 3]
```diff
@@ -3,40 +3,27 @@
...
+  if [ "$checkHttpUrl" = "" ] && [ "$checkSshUrl" = "" ]; then
```
Comment (whatwewant) [inline core/mr:14]:
```suggestion
  if [ "$check_httpUrl" = "" ] && [ "$check_ssh_url" = "" ]; then
```

**Pull request merged**